src/Reader/AppendReader.php

```diff
@@ -9,43 +11,47 @@
  *
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-final class AppendReader extends \AppendIterator implements Reader
+final class AppendReader implements Reader, CountableReader
 {
+    private $readers = [];
+
     /**
      * @param Reader[] $readers
      */
     public function __construct(array $readers = [])
     {
-        parent::__construct();
-
         foreach ($readers as $reader) {
             $this->addReader($reader);
         }
     }
 
-    /**
-     * Safety check method to make sure a Reader is passed.
-     *
-     * @param Reader $reader
-     */
-    public function addReader(Reader $reader)
+    public function getItems(): \Generator
     {
-        parent::append($reader);
+        foreach ($this->readers as $reader) {
+            yield from $reader->getItems();
+        }
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function append(\Iterator $iterator)
+    public function count()
     {
-        $this->addReader($iterator);
+        $cnt = 0;
+
+        foreach ($this->readers as $reader) {
```

Comment:
Not sure if we should use the [`nikc/iter`](https://github.com/nikic/iter) library for such case? I try to avoid adding new libraries to make the migration path as easy as possible.